### DISCUSSION TITLE
Scale window programatically in winUI3

### DISCUSSION THREAD

**Discussion by khoabui1412** [microsoft/microsoft-ui-xaml#8687] [Q&A]
Hi,
Is there a way to scale the window programatically in winUI3 app, instead of changing scale in system setting?
And how to prevent scale the window automatically if dpt changed in system setting?

**Comment by castorix** [comment]
I'm not sure what you're trying to do but you can get the scaling with [XamlRoot.RasterizationScale](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.xamlroot.rasterizationscale?view=windows-app-sdk-1.3) or [GetDpiForWindow](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getdpiforwindow) and you can get notified with [XamlRoot.Changed](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.xamlroot.changed?view=windows-app-sdk-1.3)
For an idea, I used _XamlRoot.Changed_ in a UserControl to resize a hosted Win32 control in this test sample : [ContainerPanel.xaml.cs](https://github.com/castorix/WinUI3_DesktopDuplication/blob/master/ContainerPanel.xaml.cs)

**Reply by khoabui1412** [reply]
I have the scale custom setting in my app, I want when change the scale for example to 200%, all windows will scale to 200%.
I can resize the window to 200% but it's not literally scale to 200%, and I need to implement scale for every UIelement inside.
So I want to know the mechanism in scaling app when change scale in system setting, then I can do the same.

**Comment by DHancock** [comment]
A [ScrollViewer](https://learn.microsoft.com/en-gb/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.scrollviewer?view=windows-app-sdk-1.2) control may be what you're after. Wrap any content that you want to zoom in the ScrollViewer, then to set the zoom factor call [ScrollViewer.ChangeView()](https://learn.microsoft.com/en-gb/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.scrollviewer.changeview?view=windows-app-sdk-1.2) from your code behind. There is an example in the WinUI3 Gallery app. (edit: using ZoomToFactor() isn't recommended)

**Reply by khoabui1412** [reply]
I think it's still not a correct scale, there will be conflict in resize window and scale Ui element,
In this case view can be cropped.
